Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=7dac745b8e367c99175b8f0d014d996f0e5ed9e5
Commit:     7dac745b8e367c99175b8f0d014d996f0e5ed9e5
Parent:     d1b3525b4126d7acad0493b62642b80b71442661
Author:     Tejun Heo <tj@kernel.org>
AuthorDate: Thu Feb 12 10:34:32 2009 +0900
Committer:  Jeff Garzik <jgarzik@redhat.com>
CommitDate: Mon Feb 16 18:57:25 2009 -0500

    sata_nv: give up hardreset on nf2
    
    Kernel bz#12176 reports that nf2 hardreset simply doesn't work.  Give
    up.  Argh...
    
    Signed-off-by: Tejun Heo <tj@kernel.org>
    Cc: Robert Hancock <hancockr@shaw.ca>
    Reported-by: Saro <saro_v@hotmail.it>
    Signed-off-by: Jeff Garzik <jgarzik@redhat.com>

 drivers/ata/sata_nv.c |   14 ++++++++------
 1 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/drivers/ata/sata_nv.c b/drivers/ata/sata_nv.c
index 444af04..55a8eed 100644
--- a/drivers/ata/sata_nv.c
+++ b/drivers/ata/sata_nv.c
@@ -421,19 +421,21 @@ static struct ata_port_operations nv_generic_ops = {
 	.hardreset		= ATA_OP_NULL,
 };
 
-/* OSDL bz3352 reports that nf2/3 controllers can't determine device
- * signature reliably.  Also, the following thread reports detection
- * failure on cold boot with the standard debouncing timing.
+/* nf2 is ripe with hardreset related problems.
+ *
+ * kernel bz#3352 reports nf2/3 controllers can't determine device
+ * signature reliably.  The following thread reports detection failure
+ * on cold boot with the standard debouncing timing.
  *
  * http://thread.gmane.org/gmane.linux.ide/34098
  *
- * Debounce with hotplug timing and request follow-up SRST.
+ * And bz#12176 reports that hardreset simply doesn't work on nf2.
+ * Give up on it and just don't do hardreset.
  */
 static struct ata_port_operations nv_nf2_ops = {
-	.inherits		= &nv_common_ops,
+	.inherits		= &nv_generic_ops,
 	.freeze			= nv_nf2_freeze,
 	.thaw			= nv_nf2_thaw,
-	.hardreset		= nv_noclassify_hardreset,
 };
 
 /* For initial probing after boot and hot plugging, hardreset mostly
